simulator/ns-3.35/src/point-to-point/model/switch-mmu.cc

```diff
@@ -155,11 +164,18 @@ SwitchMmu::SetReserved(uint64_t b, std::string inout) {
 	if (inout == "ingress") {
 		for (uint32_t port = 0; port < pCnt; port++) {
 			for (uint32_t q = 0; q < qCnt ; q++) {
+				if (totalIngressReserved >= reserveIngress[port][q])
+					totalIngressReserved -= reserveIngress[port][q];
```

Comment:
Why there are two "SetReserved" functions

Reply:
These are overloaded functions. 

One function sets reserved for a port, qIndex. Other function sets reserved for all queues in one-shot.